server/settings_test.go

```diff
@@ -618,20 +622,80 @@ groups:
 
 					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
 						Body: server.ChangeSettingsBody{
-							AlertManagerURL:   url,
+							AlertManagerURL:   alertManagerURL,
 							AlertManagerRules: rules,
 						},
 						Context: pmmapitests.Context,
 					})
 					require.NoError(t, err)
-					assert.Equal(t, url, res.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, res.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, res.Payload.Settings.AlertManagerRules)
 
 					gets, err := serverClient.Default.Server.GetSettings(nil)
 					require.NoError(t, err)
-					assert.Equal(t, url, gets.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, gets.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, gets.Payload.Settings.AlertManagerRules)
 
+					t.Run("PrometheusReloaded", func(t *testing.T) {
+						prometheusBaseURL := pmmapitests.BaseURL.ResolveReference(&url.URL{
+							Path: "prometheus/",
+						})
+						client, err := prometheusApi.NewClient(prometheusApi.Config{
+							Address: prometheusBaseURL.String(),
+						})
+						require.NoError(t, err)
+
+						expected := prometheusApiV1.RuleGroup{
+							Rules: prometheusApiV1.Rules{
+								prometheusApiV1.AlertingRule{
+									Alerts:      []*prometheusApiV1.Alert{},
+									Annotations: model.LabelSet{"summary": "High request latency"},
+									Duration:    600,
+									Labels:      model.LabelSet{"severity": "page"},
+									Name:        "HighRequestLatency",
+									Query:       `job:request_latency_seconds:mean5m{job="myjob"} > 0.5`,
+								},
+							},
+							File:     "/srv/prometheus/rules/pmm.rules.yml",
+							Interval: 60,
+							Name:     "example",
+						}
+
+						// Prometheus needs some time to reload configs.
+						ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+						defer cancel()
+
+						api := prometheusApiV1.NewAPI(client)
+						waitCount := 0
+						for {
+							result, err := api.Rules(ctx)
+							// Health is not reliable. It might return ok but many times it returns unknown even after a lot of retries
+							// Error 503 is Prometheus is not available yet
```

Comment:
So Prometheus crashes and we just add workarounds for tests to pass? We should not do that unless we clearly understand why this happens.

Reply:
@percona-csalguero how often did you get this error?
Is it possible that we are sending to much requests to prometheus that it's returning 503 error?
Based on prometheus API documentation we are getting 503 Service Unavailable when queries time out or abort.
https://prometheus.io/docs/prometheus/latest/querying/api/

Reply:
Are we even getting 503 from Prometheus and not nginx?

Reply:
So HTTP status code 503 doesn't mean that prometheus crashed.
Should we make reverse engineering of prometheus code to check why it might happen?

Reply:
Why not start with logs and inspecting the actual response?

Reply:
I'm checking it.

Reply:
Removed all other checks in `if` statement except 503. Ran 20 times locally and didn't get 503 error at all.